homeassistant/components/rainbird/__init__.py

```diff
@@ -1,42 +1,96 @@
 """Support for Rain Bird Irrigation system LNK WiFi Module."""
 import logging
 
+from pyrainbird import RainbirdController
 import voluptuous as vol
 
+from homeassistant.components import binary_sensor, sensor, switch
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_TRIGGER_TIME,
+)
+from homeassistant.helpers import discovery
 import homeassistant.helpers.config_validation as cv
-from homeassistant.const import CONF_HOST, CONF_PASSWORD
+
+CONF_ZONES = "zones"
+
+SUPPORTED_PLATFORMS = [switch.DOMAIN, sensor.DOMAIN, binary_sensor.DOMAIN]
 
 _LOGGER = logging.getLogger(__name__)
 
+RAINBIRD_CONTROLLER = "controller"
 DATA_RAINBIRD = "rainbird"
 DOMAIN = "rainbird"
 
-CONFIG_SCHEMA = vol.Schema(
+SENSOR_TYPE_RAINDELAY = "raindelay"
+SENSOR_TYPE_RAINSENSOR = "rainsensor"
+# sensor_type [ description, unit, icon ]
+SENSOR_TYPES = {
+    SENSOR_TYPE_RAINSENSOR: ["Rainsensor", None, "mdi:water"],
+    SENSOR_TYPE_RAINDELAY: ["Raindelay", None, "mdi:water-off"],
+}
+
+TRIGGER_TIME_SCHEMA = vol.Any(
+    cv.positive_int,
```

Comment:
It's already valid to pass integers as seconds in timedelta, in `cv.time_period`.

Reply:
Yes, but as seconds. Common irrigation time is between 1 to 30 minutes. I do not se the reason to force users multiply minutes by 60 mynually when they want to se minutes of irrigation.

Reply:
They can set minutes by using the minutes interface in a time dict or a time string.

We use the timedelta interface all over home assistant to specify time interval. Keeping a consistent interface helps users.

Reply:
Please remove this.